Cargo.toml

```diff
@@ -38,7 +38,9 @@ docopt = "1.1"
 env_logger = "0.10"
 futures = { version = "0.3", features = ["compat"] }
 hex = "0.4"
+hmac = "0.12"
 http = "0.2"
+jsonwebtoken = "9.2"
```

Comment:
Note I'm introducing a dependency here, jsonwebtoken is widely used, with over 14 Million downloads (https://crates.io/crates/jsonwebtoken).

However, jsonwebtoken does pull in [`ring`](https://crates.io/crates/ring/) directly. That might be okay, because rustls is already in our dependency tree and it depends on `ring` for crypto but wanted to make sure this is called out explicitly. There are other options to validate jwks natively in Rust if we go another direction although might be a tad bit more painful to implement